Lib/test/test_ast/test_ast.py

```diff
@@ -220,6 +220,29 @@ def test_negative_locations_for_compile(self):
                 # This also must not crash:
                 ast.parse(tree, optimize=2)
 
+    def test_docstring_optimization_single_node(self):
```

Comment:
Maybe add a test where the docstring spans multiple lines as well and does not necessarily have the same indentation as the outer body and also when we maybe have an empty line between the def and the docstring:

```py
def f1():
    """abc
    """

def f2():
    """
abc
    """

def f3():
    """
       abc
    """

def f4():
    """abc
"""


def f5():

    """abc
"""
```